Automattic/studio
# fix(sync): refetch sites on opening popup

## Related issues
- Fixes https://github.com/Automattic/dotcom-forge/issues/9584

## Proposed Changes
Previously opening Sync popup didn't trigger refetching of sites. So if a user opened "Sync" tab, then created wordpress.com site and then clicked "Connect" button in Sync tab - they didn't see recently created website, so it was necessary to reload Studio.
With this PR we are refetching sites each time the popup is opening .

## Testing Instructions
1. Run Studio with `STUDIO_SITE_SYNC=true npm start`
2. Select any site in left sidebar and open Sync tab
3. Go to wordpress.com and purchase a website
4. Come back to Studio and click on "Connect site"
5. Assert that you see just created site in the list (previously it would not be there)
Actually, I think that now the code looks even more straightforward, w/o extra props drilling and handleConnect is now isolated inside the dialog. Moreover, we *removed*  30% of lines - `+29 −45`
